src/coreclr/tools/Common/Compiler/CompilerTypeSystemContext.Validation.cs

```diff
@@ -368,6 +368,16 @@ private static TypeDesc EnsureLoadableTypeUncached(TypeDesc type)
                     ThrowHelper.ThrowTypeLoadException(ExceptionStringID.ClassLoadGeneral, type);
                 }
 
+                // Validate fields meet the constraints of the enclosing type.
```

Comment:
I think this is the kind of check we could leave out from AOT compilers. A field not meeting constraints doesn't seem any more damaging than a local or parameter and we don't check those and probably wouldn't want to spend time checking them.

Reply:
Okay.